dev/io.openliberty.faces.4.0.internal/src/io/openliberty/faces40/internal/extprocessor/JSFExtensionFactory.java

```diff
@@ -138,6 +138,22 @@ public ExtensionProcessor createExtensionProcessor(IServletContext webapp) throw
         //set a context attribute for later queries on which JSF impl is in use.
         webapp.setAttribute(JSFConstants.JSF_IMPL_ENABLED_PARAM, jsfImplEnabled);
 
+        /*
+         * In previous version of Faces/JSF this was done in the AbstractJSPExtensionFactory.createExtensionProcessor.
+         * In Faces 4.0 the Pages/JSP feature is no longer enabled as Pages/JSP support was removed from the Faces 4.0 Specification.
+         * If an AnnotationHandlerManager is not added here then when the WASCDIAnnotationInjectionProvider tries to get an instance
+         * of an AnnotationHelperManager null is returned.
+         *
+         * The WASMyFacesContainerInitializer will do this for applications that have a FacesServlet defined dynamically
```

Comment:
Shouldn't the comment say JSFExtensionFactory instead?

Reply:
Nope, the comment is correct, each comment talks about the other class so it is clear where the logic is taking place.